src/php/Command/Test/TestCommandOptions.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Command\Test;
+
+final class TestCommandOptions
+{
+    private const FILTER_OPTION = '--filter=';
+
+    private string $filter = '';
+
+    public static function empty(): self
+    {
+        return new self();
+    }
+
+    public static function fromArray(array $options): self
+    {
+        $self = new self();
+
+        foreach ($options as $option) {
+            if (false !== strpos($option, self::FILTER_OPTION)) {
+                $self->filter = str_replace('-', '_', substr($option, strlen(self::FILTER_OPTION)));
+            }
+        }
+
+        return $self;
+    }
+
+    public function asPhelHashMap(): string
```

Comment:
This looks ugly and there are probably some edge case where it will fail. Why not make a new instance of the Map class (using TypeFactory) and then use the Printer to print the result?

Reply:
I agree. Let's talk in private to agree on this. Let's sync first on the way of how do we want to handle inputs and options.